src/plugins/intel_npu/src/plugin/npuw/partitioning/online/snapshot.cpp

```diff
@@ -944,7 +1001,8 @@ bool Snapshot::cleanUpUniquesImpl(const GPtrSet& gptrs) {
     // Another special case, actually a workaround. Keep it
     // FIXME: slightly different from Ensemble since we don't check flops and keep it by size only
     auto block_layer_size = (*(gptrs.begin()))->size();
-    if (gptrs.size() >= m_ctx.keep_blocks && block_layer_size >= m_ctx.keep_block_size) {
+    if ((gptrs.size() >= m_ctx.keep_blocks && block_layer_size >= m_ctx.keep_block_size) ||
+        (gptrs.size() * block_layer_size > 500)) {
```

Comment:
Now instead of 30*96 groups we could get 7*96, 7*96, 10*79, 6*96. Please, suggest a better check here